### PR TITLE
ARTEMIS-1131 Configure optimal GC settings

### PR DESCRIPTION
**Pull request by franz1981**

### CODE REVIEW

File: docs/user-manual/en/perf-tuning.md [hunk 1]
```diff
@@ -206,6 +206,14 @@ tunings won't apply to JDKs from other providers (e.g. IBM or JRockit)
     of your queues and the size and number of your messages. Use the JVM
     arguments `-Xms` and `-Xmx` to set server available RAM. We
     recommend setting them to the same high value.
+    
+    When under periods of high load, it is likely that Artemis will be generating 
+    and destroying lots of objects. This can result in a build up of stale objects. 
+    To reduce the chance of running out of memory and causing a full GC 
+    (which may introduce pauses and unintentional behaviour), it is recommended that the 
+    max heap size (`-Xmx`) for the JVM is set at least to 5 x the `global-max-size` of the broker.
```
Comment (clebertsuconic) [inline docs/user-manual/en/perf-tuning.md]: Can you change the calculation for global-max-size then to be 1/5 of the -Xmx? 

Look on ConfigurationImpl

Comment (franz1981) [inline docs/user-manual/en/perf-tuning.md]: It is not a bad idea IMHO, but these hints are supposed to work best when a user is near 100% sure that the broker will be hammered by requests (without any expire queue) while retaining in memory the whole `global-max-size` bytes...
I've the impression that raising the requirements (5  times more!) for the common use case to satisfy the uncommon ones could be to too restrictive, wdyt?

Comment (mtaylor) [inline docs/user-manual/en/perf-tuning.md]: +1 @franz1981 I think we already discussed this on IRC but, having the global-max-size being 1/5th of heap by default for the out of the box configuration seems like overkill.  These settings are meant to target a particular use case.

Comment (mtaylor) [inline docs/user-manual/en/perf-tuning.md]: I'm going ahead and merging this.  We can continue discussion about sensible defaults and send a separate patch.
